src/overlays/actors/ovl_En_Stone_heishi/z_en_stone_heishi.c

```diff
@@ -35,54 +39,473 @@ const ActorInit En_Stone_heishi_InitVars = {
     (ActorFunc)EnStoneheishi_Draw,
 };
 
-// static ColliderCylinderInit sCylinderInit = {
-static ColliderCylinderInit D_80BCA3A0 = {
-    { COLTYPE_NONE, AT_NONE, AC_NONE, OC1_ON | OC1_TYPE_ALL, OC2_TYPE_2, COLSHAPE_CYLINDER, },
-    { ELEMTYPE_UNK0, { 0x00000000, 0x00, 0x00 }, { 0xF7CFFFFF, 0x00, 0x00 }, TOUCH_NONE | TOUCH_SFX_NORMAL, BUMP_NONE, OCELEM_ON, },
+static ColliderCylinderInit sCylinderInit = {
+    {
+        COLTYPE_NONE,
+        AT_NONE,
+        AC_NONE,
+        OC1_ON | OC1_TYPE_ALL,
+        OC2_TYPE_2,
+        COLSHAPE_CYLINDER,
+    },
+    {
+        ELEMTYPE_UNK0,
+        { 0x00000000, 0x00, 0x00 },
+        { 0xF7CFFFFF, 0x00, 0x00 },
+        TOUCH_NONE | TOUCH_SFX_NORMAL,
+        BUMP_NONE,
+        OCELEM_ON,
+    },
     { 15, 70, 0, { 0, 0, 0 } },
 };
 
-#endif
+static u16 sEnStoneHeishiTextIds[] = { 0x1473, 0x1474, 0x1475, 0x1476, 0x1477, 0x1478, 0x1479, 0x147A, 0x1472 };
 
-extern ColliderCylinderInit D_80BCA3A0;
+static AnimationHeader* sEnStoneHeishiAnimationInfoS[] = {
+    &gSoldierStandHandOnHip, &gSoldierDrink, &gSoldierCheerWithSpear, &gSoldierWave,
+    &gSoldierSitAndReach,    &gSoldierDrink, &gSoldierStandUp,
+};
+
+static u8 sEnStoneHeishiAnimations[] = { 0, 2, 0, 0, 0, 0, 2 };
+
+void EnStoneheishi_Init(Actor* thisx, PlayState* play) {
+    EnStoneheishi* this = THIS;
+
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 25.0f);
+    SkelAnime_InitFlex(play, &this->skelAnime, &gSoldierSkel, &gSoldierWave, this->jointTable, this->morphTable,
+                       SOLDIER_LIMB_MAX);
+
+    this->actor.colChkInfo.mass = MASS_IMMOVABLE;
+    this->actor.targetMode = 6;
+    this->actor.gravity = -3.0f;
+
+    Collider_InitAndSetCylinder(play, &this->collider, &this->actor, &sCylinderInit);
+
+    this->collider.dim.radius = 0x28;
+    this->collider.dim.height = 0x28;
+    this->collider.dim.yShift = 0;
+
+    func_80BC94B0(this);
+}
+
+void EnStoneheishi_Destroy(Actor* thisx, PlayState* play) {
+    EnStoneheishi* this = THIS;
+
+    Collider_DestroyCylinder(play, &this->collider);
+}
+
+void EnStoneheishi_ChangeAnim(EnStoneheishi* this, s32 animIndex) {
+    f32 startFrame = 0.0f;
+
+    this->animIndex = animIndex;
+    this->endFrame = Animation_GetLastFrame(sEnStoneHeishiAnimationInfoS[animIndex]);
+
+    if (animIndex == EN_STONE_HEISHI_DRINK_2) { // This will never pass since this animation index is never used.
```

Comment:
```suggestion

    // This will never pass since this animation index is never used.
    if (animIndex == EN_STONE_HEISHI_DRINK_2) {
```
I think the comment would look nicer above